### ISSUE TITLE
Add force option to artifact:deploy command

### ISSUE THREAD

**Issue by richardbporter** [acquia/blt#4143]
**Is your feature request related to a problem? Please describe.**
We're running into an issue where TravisCI is unable to push the build artifact to one of our git remotes due to upstream changes, even though there shouldn't be any. This results in a failing build, however it still deploys to some git remotes.

**Describe the solution you'd like**
It would be nice if there was an option to execute a force push without having to completely override the deploy command. This option would skip merging upstream changes and add the `--force` option to the `git push` exec. This would treat CI as the authoritative source which IMO should be the default anyway. Developers shouldn't be pushing directly to Acquia remotes unless in rare occasions where CI is down. 

This option would resolve the pseudo-failing build mentioned above and avoid failures where CI is unable to resolve a merge conflict.

**Describe alternatives you've considered**
I don't know of a way to add this option to our setup without completely overriding the artifact:deploy command which I'd like to avoid.

**Additional context**
I've no idea why there are upstream changes in the first place. One of our SVN instances relaunched around the time this started happening.

**Comment by richardbporter** [comment]
This is happening on BLT 11.4.1. When I build the artifact locally and replicate the failure, `git pull` says `warning: no common commits` and `fatal: refusing to merge unrelated histories` for the remote in question.

**Comment by danepowell** [comment]
If there are unexpected upstream changes, it would seem to indicate either a problem with your repository or a bug in BLT. In this case I'm leaning towards a problem with you repo, but regardless, a `--force` option seems like a dangerous solution. It would be better to diagnose the root cause here and fix that.

If there are no common commits between your local and upstream repos, then it seems like something is going wrong with the initial clone or branch checkout. Can you use `git log` to compare the local and upstream repos? They should be identical prior to BLT creating a new deploy commit.

**Comment by richardbporter** [comment]
I guess I don't understand the use case for having CI pull upstream changes (RA maybe?). It could potentially result in merge conflicts which would fail the deployment step.

If CI is being used, it seems ideal to have it be a one-way street.

**Comment by richardbporter** [comment]
> no common commits between your local and upstream repos

I think it had something to do with the SVN relaunch. I had to force push locally to get it all worked out. AC ticket 956773 for reference.

**Comment by danepowell** [comment]
I'm having a hard time imagining how BLT could cause the error you mentioned. You say you get an error running `git pull`, but BLT doesn't ever run `git pull`.

BLT inits an empty Git repo in the `deploy` directory, then checks out a new empty branch, then fetches and merges _just_ that one branch from the remote repo. It's merging into an empty local branch, so their shouldn't be conflicts. It does this instead of a full `git clone` / `git pull` for performance reasons, we don't need the whole repo in order to push one branch.

So CI _is_ a one-way street for BLT. But that doesn't mean that BLT is going to blindly blow away everything in your artifact repo with a `--force` push, that would be far too dangerous, and I don't think I even want to allow people to do that optionally.

**Comment by richardbporter** [comment]
Fair enough. Although, a force push for that one branch wouldn't blow away _everything_ in your artifact repo, just that branch. :) 

I don't think I can replicate this now, unfortunately. You're right though in that is most likely an issue with the repo. If it happens after another SVN relaunch, I'll contact Acquia Support again.

Thanks!

**Comment by danepowell** [comment]
Thanks for following up, if it happens again and you can provide steps to reproduce let us know.

**Comment by richardbporter** [comment]
FWIW, I just ran into a similar issue when adding a new git remote to our blt.yml file. The deploy step of the build errored with `! [remote rejected]   master-build -> master-build (shallow update not allowed)`.

Tried to fix similarly to what I did last time. I ran `blt artifact:build --dry-run` locally, CD'ed into the deploy.dir, unshallowed the repo using the remote that BLT built the artifact from and then finally pushed to the new remote (no force required). Re-running the build fails with `! [rejected]          master-build -> master-build (fetch first)`.

I must be missing something obvious here. Any tips?
